edb/tools/gen_meta_grammars.py

```diff
@@ -0,0 +1,169 @@
+#
+# This source file is part of the EdgeDB open source project.
+#
+# Copyright 2019-present MagicStack Inc. and the EdgeDB authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+import click
+import collections
+import datetime
+import sys
+
+from edb.edgeql.parser.grammar import keywords as eql_keywords
+from edb.eschema.parser.grammar import keywords as esc_keywords
+from edb.schema import schema as s_schema
+from edb.tools.edb import edbcommands
+
+import edgedb
+
+
+BOOL_LITERALS = {'true', 'false'}
+CONTAINER_TYPES = {'array', 'tuple'}
+NAMES = ('edgeql', 'eschema')
+NAVIGATION = ('.<', '.>', '@', '.')
+
+
+def die(msg):
+    print(f'FATAL: {msg}', file=sys.stderr)
+    sys.exit(1)
+
+
+def gen_grammar_class(name, components):
+    code = [f'class {name}:\n']
+    for varname, values in components.items():
+        code.append(f'    {varname} = (\n')
+        code.extend([f'        "{val}",\n' for val in values])
+        code.append(f'    )\n')
+
+    return code
+
+
+def main(names):
+    code = [
+        f'# AUTOGENERATED BY EdgeDB WITH\n'
+        f'#     $ edb gen-pygments-grammars {" ".join(names)}\n'
+        f'#\n'
+        f'# TIMESTAMP: {datetime.datetime.now():%Y-%m-%d %H:%M:%S}\n'
+        f'\n\n'
+    ]
+
+    con = edgedb.connect(user='edgedb', database='edgedb')
```

Comment:
`con` should be created in `gen_meta_grammars()`; `main()` should be called in `try..finally` with `con.close()` in the `finally` block

Reply:
OK, I'll update it.